src/env/feedback_env.py

```diff
@@ -79,8 +79,14 @@ def step(self, action):
 
         # check if max steps reached
         self.steps_taken += 1
-        if self.max_steps is not None and self.steps_taken >= self.max_steps:
+        if (
+            self.max_steps is not None
+            and self.steps_taken >= self.max_steps
+            and not terminated
```

Comment:
for cases where episodes are terminated at the last step

Reply:
v good catch thanks